go.mod

```diff
@@ -1,10 +1,12 @@
 module github.com/bottlerocket/hotdog
 
-go 1.15
```

Comment:
I think we should align with the golang toolchain version baked in the `bottlerocket-sdk`. 
https://github.com/bottlerocket-os/bottlerocket-sdk/blob/907998cf3f8a4093868042ba15fea8cf7c0db278/Dockerfile#L478

I'm not sure what the consequences of leaping ahead of the toolchain version in the SDK is when building Bottlerocketl; better safe than sorry.

Reply:
> toolchain version baked in the bottlerocket-sdk.

Right - I assumed we'd be on the latest Go with the SDK bumps, and I started a convo on this here: https://github.com/bottlerocket-os/bottlerocket-sdk/pull/82#discussion_r974660024

I'll draft this PR until I get clarification from Ben on why we aren't using Go 1.19+

Reply:
> I'm not sure what the consequences of leaping ahead of the toolchain version in the SDK is when building Bottlerocketl; better safe than sorry.

One of the nice things about using Go and the `go build` toolchain is they make every effort to be backwards compatible with the many versions of Go `1.x.x`. For a simple program like `hotdog` I'd say this is very low risk.

Reply:
Go 1.15 has been EOL for over a year now, so we should definitely update to something newer. Go 1.18 is a currently supported version, but not really for much longer. Major version upgrades of the Go toolchain are usually fairly safe, so I think it would be good to update to the latest and greatest if possible - for both this and the SDK.